### PR TITLE
docs: document schemas, switch to redoc

### PR DESCRIPTION
**Pull request by lidel**
This PR improves API docs:

- adds Schemas section and improves the order of sections to make onboarding easier
- switches to ReDoc
  - old doc renderer was too limited and had bugs which made it painful to create presentable docs we need
  - used by big projects such as https://docs.discourse.org or https://docs.docker.com/engine/api/v1.25/
  - enables us to pass custom YAML via `#specUrl=` for previews like below one

**PREVIEW:** https://bafybeih5udiukicsgnq65ufkkmdcsel5ojp25axfws7ifmdy52b5hfgkpi.ipfs.dweb.link/#specUrl=https://raw.githubusercontent.com/ipfs/pinning-services-api-spec/docs/switch-to-redoc/ipfs-pinning-service.yaml

### CODE REVIEW

File: ipfs-pinning-service.yaml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
   version: "0.0.3"
-  title: '(WIP) IPFS Pinning Service API'
+  title: 'IPFS Pinning Service API'
+  x-logo:
+    url: "https://bafybeidehxarrk54mkgyl5yxbgjzqilp6tkaz2or36jhq24n3rdtuven54.ipfs.dweb.link/?filename=ipfs-pinning-service.svg"
   description: "
   
   
-# Goal
+## About this spec
```
Comment (lidel) [inline ipfs-pinning-service.yaml:10]: @jessicaschilling mind quickly glancing at the new copy added in this PR? Thanks!

File: ipfs-pinning-service.yaml [hunk 2]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
 
-# The pin object lifecycle
+# Schemas
 
+This section describes the most important object types and conventions.
 
-## Creating a new pin object
 
-The user creates a pin object via `POST /pins` and receives a `PinStatus` response:
+Full list of fields and schemas can be found in `schemas` section of the YAML file.
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:31]:
```suggestion
A full list of fields and schemas can be found in the `schemas` section of the YAML file.
```

File: ipfs-pinning-service.yaml [hunk 3]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
 
-- `id` in `PinStatus` is `cid-of-pin-object`, which can can be used for modifying and/or removing the pin in the future
+## Pin and its status
+
+### Pin object
+
+`Pin` object is a representation of a pin request.
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:37]:
```suggestion
The `Pin` object is a representation of a pin request.
```

File: ipfs-pinning-service.yaml [hunk 4]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
 
+This section describes the most important object types and conventions.
 
-## Creating a new pin object
 
-The user creates a pin object via `POST /pins` and receives a `PinStatus` response:
+Full list of fields and schemas can be found in `schemas` section of the YAML file.
 
-- `id` in `PinStatus` is `cid-of-pin-object`, which can can be used for modifying and/or removing the pin in the future
+## Pin and its status
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:33]:
```suggestion
## Pin objects and pin status objects
```

File: ipfs-pinning-service.yaml [hunk 5]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
+
+### Pin object
+
+`Pin` object is a representation of a pin request.
+
+
+It includes `cid` of data to be pinned and optional metadata in `providers` and `meta`.
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:40]:
```suggestion
It includes the `cid` of data to be pinned and optional metadata in `providers` and `meta`.
```

File: ipfs-pinning-service.yaml [hunk 6]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
+
+
+### Pin status object
+
+`PinStatus` object is a representation of a pinning operation.
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:45]:
```suggestion
The `PinStatus` object is a representation of a pinning operation.
```

File: ipfs-pinning-service.yaml [hunk 7]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
+
+It includes original `pin` object along with current `status` and `id` which which can be used for status checks and management.
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:47]:
```suggestion
It includes the original `pin` object, along with current `status` and `id`, which which can be used for status checks and management.
```

File: ipfs-pinning-service.yaml [hunk 8]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
+
+
+## Pin lifecycle
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:50]:
```suggestion
## The pin lifecycle
```

File: ipfs-pinning-service.yaml [hunk 9]
```diff
@@ -14,46 +16,69 @@ The IPFS Pinning Service API is intended to be an implementation-agnostic API&#x
...
+
+
+### Creating a new pin object
+
+The user sends one or more `Pin` objects via `POST /pins` and receives `PinStatus` response for each:
+
+- `id` in `PinStatus` is `cid-of-pin-object`, which can can be used for checking status, modifying and/or removing the pin in the future
```
Comment (jessicaschilling) [inline ipfs-pinning-service.yaml:57]:
```suggestion
- `id` in `PinStatus` is `cid-of-pin-object`, which can can be used for checking status, modifying the pin, and/or removing the pin in the future
```

File: ipfs-pinning-service.yaml (hunk 4 above)
Comment (lidel) [reply ipfs-pinning-service.yaml]: @jessicaschilling I'm afraid its bit too long and gets wrapped in the menu, anything shorter we can use? Maybe just "Pin objects" ?

>

Comment (jessicaschilling) [reply ipfs-pinning-service.yaml]: "Objects" is great, thanks!

**Pull request merged**